api/v1/users/PKPUserHandler.inc.php

```diff
@@ -279,4 +284,56 @@ private function _processAllowedparams($params, $allowedKeys) {
 
 		return $returnParams;
 	}
+
+	/**
+	 * Retrieve the user report
+	 *
+	 * @param Slim\Http\Request $slimRequest Slim request object
+	 * @param \APIResponse $response Response
+	 * @param array $args
+	 * @return \APIResponse Response
+	 */
+	public function getReport(\Slim\Http\Request $slimRequest, \APIResponse $response, array $args): \APIResponse
+	{
+		$request = $this->getRequest();
+
+		$context = $request->getContext();
+		if (!$context) {
+			return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+		}
+
+		$params = ['contextId' => $context->getId()];
+		foreach ($slimRequest->getQueryParams() as $param => $value) {
+			switch ($param) {
+				case 'userGroupIds':
+					if (is_string($value) && strpos($value, ',') > -1) {
+						$value = explode(',', $value);
+					} elseif (!is_array($value)) {
+						$value = [$value];
+					}
+					$params[$param] = array_map('intval', $value);
+					break;
+				case 'mappings':
+					if (is_string($value) && strpos($value, ',') > -1) {
+						$value = explode(',', $value);
+					} elseif (!is_array($value)) {
+						$value = [$value];
+					}
+					$params[$param] = $value;
+					break;
+			}
+		}
+
+		\HookRegistry::call('API::users::user::report::params', [&$params, $slimRequest]);
+
+		$report = \Services::get('user')->getReport($params);
+		$output = fopen(tempnam(sys_get_temp_dir(), 'tmp'), 'w+');
```

Comment:
Is it possible to use `fopen('php://stdout', 'w')` or something to avoid a temporary file? The only challenge woudl be getting the headers to be sent first, the way Smarty's response chaining works.

Reply:
Yeah, I had an intention here (related to blocking the script under slow connections), but it was based in an old myth from my head, I've just killed it a test =]

Reply:
I was taking a look at the Slim implementation and they are using `php://temp` to buffer the output, so I've just replaced by it, with some good luck, the file system won't be used...

Also, I've tried to skip the Slim conventions, by just writing to the stdout and exiting, but looks it's using the ob_* functions. I'll check how to disable the buffering 🤔

Reply:
OK, thanks! That works for now, but if/when either of us get the chance it would be good to tinker with ways to let both coexist without violence.

Reply:
@asmecher Oh, I was waiting for a comment before the merge :)

For example, I think the code below would work, due to the Slim buffer, but it would end up writing to a temp file indirectly:

```php
$report->serialize(fopen('php://output', 'w+'));
return $response
	->withHeader('content-type', 'text/comma-separated-values')
	->withHeader('content-disposition', 'attachment; filename="user-report-' . date('Y-m-d') . '.csv"');
```